jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ClientConnector.java

```diff
@@ -753,4 +833,44 @@ public ChannelWithAddress newChannelWithAddress(ClientConnector clientConnector,
             };
         }
     }
+    
+    /**
+     * <p>A listener for events about TCP connection establishment and failure.</p>
+     * <p>The events:</p>
+     * <ul>
+     * <li>Beginning of connect attempt</li>
+     * <li>Success of connect attempt</li>
+     * <li>Failed connect attempt</li>
+     * </ul>
+     */
+    public interface ConnectListener extends EventListener
```

Comment:
As this is a supplement to `Connection.Listener`, we should do something to alleviate confusion if possible:

I.e.: a `ClientConnector.ConnectListener` is not a client version of `Connection.Listener` like the names alone could suggest. Maybe calling it `TcpConnectListener` and adding some javadoc referencing `Connection.Listener` would help?